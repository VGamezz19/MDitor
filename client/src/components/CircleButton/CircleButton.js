import React from 'react';
import { Button } from 'react-materialize'

function CircleButton(props) {

    const { className, icon, onClick  } = props

    const handlerOnClick = () => {
        if (onClick) {
            return onClick()
        }
    }
    
    return (
        <Button
            floating large
            className={`main-icon ${className}`}
            onClick={handlerOnClick}
            icon={icon}
            waves='light'>
        </Button>
    )

}

export default CircleButton