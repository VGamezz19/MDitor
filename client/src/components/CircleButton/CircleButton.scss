@import '../../sass/_variables';
@import '../../sass/_extends';
.add-folder,
.add-files {
    align-self: center;
    margin-top: 1em;
    margin-bottom: 1em;
}

.main-icon {
    i {
        @extend %medium-main-icon;
    }
}