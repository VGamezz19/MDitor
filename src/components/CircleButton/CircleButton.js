import React, { Component } from 'react';
import { Button, Icon } from 'react-materialize'

class CircleButton extends Component {
    constructor(props) {
        super(props)
        this.state = {

        }
    }

    handlerOnClick = () => {
        console.log("circle clicked")
        this.props.onClick()
    }

    render() {
        return (
            <Button 
                floating large 
                className={this.props.ButtonClassName} 
                onClick={this.handlerOnClick}
                waves='light'>
                <Icon className={this.props.IconClassName + " main-icon"}> {this.props.icon}</Icon>
            </Button>
        )
    }
}

export default CircleButton