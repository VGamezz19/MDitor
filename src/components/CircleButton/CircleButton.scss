.black-font {
    color:black !important;
}