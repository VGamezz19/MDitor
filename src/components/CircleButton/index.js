import CircleButton from './CircleButton'
import './CircleButton.scss'

export default CircleButton