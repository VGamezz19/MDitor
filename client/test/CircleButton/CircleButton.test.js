import React from 'react';
import ReactDOM from 'react-dom';
import CircleButton from '../../src/components/CircleButton';
import renderer from 'react-test-renderer';

it('CircleButton should rendering', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CircleButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Shoult create new Snapshot with this props', () => {
  const component = renderer.create(
    <CircleButton className={'grey lighten-2'} iconClassName= 'black-font' icon= 'remove_red_eye' onClick={ () => console.log("Hello Cosmos!")} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});