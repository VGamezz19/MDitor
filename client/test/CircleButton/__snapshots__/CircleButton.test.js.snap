// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoult create new Snapshot with this props 1`] = `
<button
  className="btn waves-effect waves-light btn-large btn-floating main-icon grey lighten-2"
  disabled={false}
  icon="remove_red_eye"
  onClick={[Function]}
>
  <i
    className="material-icons"
  >
    remove_red_eye
  </i>
</button>
`;
